lib/dialog-renderer/index.tsx

```diff
@@ -1,64 +1,87 @@
-import React, { Fragment } from 'react';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
 import Modal from 'react-modal';
 import PropTypes from 'prop-types';
 import classNames from 'classnames';
+import { get } from 'lodash';
 
-import * as Dialogs from '../dialogs';
+import AboutDialog from '../dialogs/about';
+import ImportDialog from '../dialogs/import';
+import SettingsDialog from '../dialogs/settings';
+import ShareDialog from '../dialogs/share';
 
-export const DialogRenderer = props => {
-  const {
-    appProps,
-    buckets,
-    themeClass,
-    closeDialog,
-    dialogs,
-    isElectron,
-    isMacApp,
-  } = props;
+import * as S from '../state';
+import * as T from '../types';
 
-  const renderDialog = dialog => {
-    const { key, title } = dialog;
-    const DialogComponent = Dialogs[dialog.type];
+type StateProps = {
+  dialogs: T.DialogType[];
+};
+
+type DispatchProps = {
+  closeDialog: () => any;
+};
 
-    if (DialogComponent === null) {
-      throw new Error('Unknown dialog type.');
-    }
+type Props = StateProps & DispatchProps;
 
-    const closeThisDialog = () => closeDialog({ key });
+export class DialogRenderer extends Component<Props> {
+  static displayName = 'DialogRenderer';
+
+  static propTypes = {
+    appProps: PropTypes.object.isRequired,
+    buckets: PropTypes.object,
+    themeClass: PropTypes.string,
+    isElectron: PropTypes.bool.isRequired,
+    isMacApp: PropTypes.bool.isRequired,
+  };
+
+  render() {
+    const {
+      appProps,
+      buckets,
+      themeClass,
+      isElectron,
+      isMacApp,
+      closeDialog,
+    } = this.props;
 
     return (
-      <Modal
-        key={key}
-        className="dialog-renderer__content"
-        contentLabel={title}
-        isOpen
-        onRequestClose={closeThisDialog}
-        overlayClassName="dialog-renderer__overlay"
-        portalClassName={classNames('dialog-renderer__portal', themeClass)}
-      >
-        <DialogComponent
-          buckets={buckets}
-          dialog={dialog}
-          requestClose={closeThisDialog}
-          isElectron={isElectron}
-          isMacApp={isMacApp}
-          {...appProps}
-        />
-      </Modal>
+      <Fragment>
+        {this.props.dialogs.map(dialog =>
+          'ABOUT' === dialog ? (
+            <AboutDialog key="dialog-renderer__about" themeClass={themeClass} />
```

Comment:
keys _only_ need to be unique up to the closest parent, so we are free to drop the prefix here - I think, unless it yelled at you because `<Fragment>` isn't a real parent.

```ts
key="about"
```